### PR TITLE
integrate new metadata schema (master)

### PR DESCRIPTION
**Pull request by d-w-moore**
Contingent on 
  - final review and merge of #64
  - successful tests with elasticlient-0.1.0-2 master as pulled from package repo.

### CODE REVIEW

**Comment by d-w-moore** [comment]
Just discovered the plugin is not working on master.  Debugging ... may have more details tomorrow (tue).

File: configuration.cpp [hunk 1]
```diff
@@ -20,13 +22,17 @@ namespace irods {
                 capture_parameter("url_template", urlTemplate);
                 capture_parameter("maximum_delay_time", maximum_delay_time);
                 capture_parameter("delay_parameters",   delay_parameters);
-            } catch ( const boost::bad_any_cast& _e ) {
-                THROW( INVALID_ANY_CAST, _e.what() );
-            } catch ( const exception _e ) {
+            } catch ( const exception& _e ) {
                 THROW( KEY_NOT_FOUND, _e.what() );
+            } catch ( const nlohmann::json::exception& _e ) {
+                rodsLog(LOG_ERROR, "JSON error -- Function %s  Line %d",__func__,__LINE__);
```
Comment (trel) [inline configuration.cpp:28]: let's pick a consistent way of printing these

File: plugin_specific_configuration.cpp [hunk 2]
```diff
@@ -17,15 +20,24 @@ namespace irods {
...
+            } catch ( const nlohmann::json::exception& _e ) {
+                rodsLog(LOG_ERROR, "JSON error -- Function %s  Line %d",__func__,__LINE__);
+                THROW( SYS_LIBRARY_ERROR, _e.what() );
+
+            } catch ( const std::exception& e ) {
+                rodsLog(LOG_ERROR, "General exception in %s - %s", __func__, e.what());
+                THROW(SYS_INTERNAL_ERR, e.what());
+
+            } catch ( ... ) {
+                THROW( SYS_UNKNOWN_ERROR, fmt::format( "Function {} File {} Line {}",__func__,__FILE__,__LINE__));
```
Comment (trel) [inline plugin_specific_configuration.cpp:40]: same, want consistency for these

**Review by alanking — COMMENTED** [review]
Logging comments

File: configuration.cpp [hunk 3]
```diff
@@ -10,22 +12,27 @@ namespace irods {
...
                 capture_parameter("index", index);
                 capture_parameter("minimum_delay_time", minimum_delay_time);
+                capture_parameter("url_template", urlTemplate);
                 capture_parameter("maximum_delay_time", maximum_delay_time);
                 capture_parameter("delay_parameters",   delay_parameters);
-            } catch ( const boost::bad_any_cast& _e ) {
-                THROW( INVALID_ANY_CAST, _e.what() );
-            } catch ( const exception _e ) {
+            } catch ( const exception& _e ) {
                 THROW( KEY_NOT_FOUND, _e.what() );
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply configuration.cpp:26]: Thanks Alan - yes, it's probably a neglected corner of programming in general , testing out the exception messages.

**Comment by d-w-moore** [comment]
Now builds, and passes the test suite, on **master** branch.

**Comment by trel** [comment]
excellent - please address the comments above, and then squash to only commits with issues attached.

**Comment by d-w-moore** [comment]
Ok,, I can do that here in this branch, and then backport later to make a 4.2.10.2 or 4.2.11.0  release, if that seems good.

**Comment by trel** [comment]
yes, let's get this code ready for merge...  and anything else you want to address, please open new issues and tackle them in separate PRs.

**Comment by trel** [comment]
see the edits in #78 to get the CMake variables consistent in this PR too.

**Comment by trel** [comment]
yes, please # these

**Comment by d-w-moore** [comment]
#'d and ready for merge !

**Pull request merged**